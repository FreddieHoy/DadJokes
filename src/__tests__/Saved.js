

import React from 'react'
import { shallow } from 'enzyme'

import SavedJokes from '../components/pages/Saved'


it('should render correctly with no props', () => {
  const component = shallow(<SavedJokes/>)
  expect(component).toMatchSnapshot()
})

it('should render banner text correctly with given strings', () => {
  const strings = ['one', 'two']
  const component = shallow(<SavedJokes list={strings} />)
  expect(component).toMatchSnapshot()
})


it('should delete all jokes when not funny dad is clicked', () => {
  const wrapper = shallow(<SavedJokes />)
  const jokes = wrapper.find('p').text()
  const button = wrapper.find('button')
  button.simulate('click')

  expect(jokes).toEqual(' - ')
})
