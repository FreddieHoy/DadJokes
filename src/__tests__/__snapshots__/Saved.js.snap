// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render banner text correctly with given strings 1`] = `
<div
  id="saved"
>
  <h2>
    what did you say dad?..
  </h2>
  <div>
    <div>
      <p>
         - 
      </p>
    </div>
  </div>
  <button
    onClick={[Function]}
  >
    DAD YOU'RE NOT FUNNY!
  </button>
  <h4>
    (CAUTION)
  </h4>
</div>
`;

exports[`should render correctly with no props 1`] = `
<div
  id="saved"
>
  <h2>
    what did you say dad?..
  </h2>
  <div>
    <div>
      <p>
         - 
      </p>
    </div>
  </div>
  <button
    onClick={[Function]}
  >
    DAD YOU'RE NOT FUNNY!
  </button>
  <h4>
    (CAUTION)
  </h4>
</div>
`;
